api/src/prescription/organization-learner/infrastructure/repositories/organization-learner-repository.js

```diff
@@ -74,14 +74,23 @@ async function get(organizationLearnerId) {
   throw new NotFoundError(`Student not found for ID ${organizationLearnerId}`);
 }
 
-async function findPaginatedLearners({ organizationId, page }) {
+async function findPaginatedLearners({ organizationId, page, filter }) {
   const query = knex
     .select('id', 'firstName', 'lastName', 'organizationId', 'attributes')
     .from('view-active-organization-learners')
     .where({ isDisabled: false, organizationId })
     .orderBy('firstName', 'ASC')
     .orderBy('lastName', 'ASC');
 
+  if (filter) {
+    Object.entries(filter).forEach(([name, values]) => {
+      query.andWhere(function () {
+        // eslint-disable-next-line knex/avoid-injections
+        this.whereRaw(`attributes->>'${name}' in ( ${values.map((_) => '?').join(' , ')} )`, values);
```

Comment:
**suggestion**: Est ce que ça ça fonctionnerait pour votre usage ? https://knexjs.org/guide/query-builder.html#wherejsonobject

```
.whereJsonPath('attributes', `'$.${name}'`, 'in', values);`
```

Reply:
on éviterait le ligne knex/avoid-injections

Reply:
On avait testé hier et retesté avec ce que tu proposes et on a ça comme résultat
`TypeError: The operator "in" is not permitted`

Reply:
`knex.raw('in')` alors . Knex me prend le chou avec leur opérateur non autorisé bien que valide 🤣.

Ou alors c'est qu'il écrit la requête de manière peu conventionnelle 😅